### DISCUSSION TITLE
Two macOS problems and their solutions

### DISCUSSION THREAD

**Discussion by codeitlikemiley** [neovide/neovide#1984] [General]
Ive been Using Neovide for Rust Development

I've used it on all platforms , but mainly on my Mac M1

Common Issues that you guys will face

1. CMD + H , cant be bind on Mac  if it is launch using `neovide` cli 

Here is how I bind CMD + H  and CMD + Q to other Keys

<img width="669" alt="Screenshot 2023-08-18 at 11 40 08 AM" src="https://github.com/neovide/neovide/assets/28816690/71aa0adb-829f-48ce-a9bd-dcd014555f39">

Solution:  create Alias neovide , paste it on your zshrc/bashrc 

```sh
neovide() {
	local arg="${1:-.}"
	local neovide_cwd
	local file

	if [[ -d "$arg" ]]; then
		neovide_cwd="$(realpath "${arg}")"
		file="$neovide_cwd"
	else
		file="$(realpath "${arg}")"
		neovide_cwd="$(dirname "${file}")"
	fi

	NEOVIDE_CWD="$neovide_cwd" open -a Neovide.app --args "$file" --multigrid --frame buttonless
}
```

3. <strong>PATH<strong> and <strong>CWD<strong> is  <strong>nil</strong> 

This happens since Neovide runs on Sandbox on MacOS

The above snippets only fixes half of the problem.

<strong>CWD<strong> gets initialized is <strong>nil<strong>  , you can check by running this

```sh
:lua print(vim.fn.getcwd())
```

So Most Plugins , Like Neo-Tree, Terminal etc that uses the vim.fn.getcwd()   have issues your opening on the root folder

<img width="345" alt="Screenshot 2023-08-16 at 9 43 43 PM" src="https://github.com/neovide/neovide/assets/28816690/d1a6ca15-41e5-4189-af3a-6507de564113">

#### For us to fix that issue...  

- prefix our command with Neovide.app with <strong>NEOVIDE_CWD</strong> , which we already did on `neovide` alias
- set our CWD on our Vim config

If using neovim with lua scripts, add the snippet on your initialization of the script , ie. `lazy.lua`

```sh
-- only set if NEOVIDE_CWD is set
if vim.fn.exists("neovide") == 1 and vim.fn.getenv("NEOVIDE_CWD") then
  vim.fn.chdir(vim.fn.getenv("NEOVIDE_CWD"))
end
```

For normal vim/nvim script

```sh
if exists("g:neovide") && !empty($NEOVIDE_CWD)
  " Change the current working directory to the value of NEOVIDE_CWD
  execute "cd " . $NEOVIDE_CWD
endif
```

If your Rust users like me on mac, you can simple use my Nvim Configuration...

https://github.com/codeitlikemiley/nvim

That allows you to move quickly around panes,  test and debug with Rust at ease...

Note all those Extra key bindings on rust only triggers if on rust project.

I think my config is 100% Better than with VSCODE with Rust :)

Hope you like it and Star thanks

**Comment by MultisampledNight** [comment]
Hi, thanks! I'll convert this into a discussion if you don't mind, since it's not a real "issue" as in the fixable sense.

Sidenote: I'm not entirely sure what the first snippet does. It binds `FILE`, but never uses it, and tries to put the _whole_ output of `env` into the environment variable `PATH`.

**Reply by codeitlikemiley** [reply]
Ive updated the snippets above, to include $FILE usage , its late night here when i wrote that issue before so i forgot to update that alias.

Also Fix an issue of getting `NEOVIDE_CWD` on plain neovim

I have added a check for neovide if exist before i get `NEOVIDE_CWD` since that would throw error if you open up neovim.

Lastly $PATH is passed , to avoid issue I had with missing Aliases , Command not Found.

You can omit that part if you wont be using a `Terminal` inside vim.
